core/src/edu/asu/c3simulator/util/Product.java

```diff
@@ -44,7 +47,7 @@ private Image getImage(String imagePath)
 	
 	public Image getProductImage()
 	{
-		return productImage;
+		return getImage(imagePath);
```

Comment:
This looks really inefficient. What is the reason for this change? Do we need the image path? Do we need unique copies of the image?